backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/diagram/ViewDiagramDescriptionConverter.java

```diff
@@ -430,6 +435,24 @@ private Function<VariableManager, List<?>> getSemanticElementsProvider(org.eclip
         };
     }
 
+    private boolean isUnsynchronizedNodeDescription(org.eclipse.sirius.components.view.DiagramElementDescription elementDescription) {
+        return elementDescription instanceof org.eclipse.sirius.components.view.NodeDescription
+                && elementDescription.getSynchronizationPolicy().getName().equals(SynchronizationPolicy.UNSYNCHRONIZED.toString());
+    }
+
+    private List<?> getUnsynchronizedSemanticElements(VariableManager variableManager, org.eclipse.sirius.components.view.NodeDescription nodeDescription) {
+        String[] semanticElementIds = variableManager.get(NodeComponent.SEMANTIC_ELEMENT_IDS, String[].class).orElse(new String[] {});
```

Comment:
If the variable was a list, you wouldn't need to convert it back to an array

Reply:
There would be a Type safety warning when trying to directly assign the variable as a List<String> with a expected type of List.class so I thought it would be cleaner to put an array which does not trigger a warning.

Reply:
You wouldn't have a warning with a code like that:

```
Object value = variableManager.getVariables().get(NodeComponent.SEMANTIC_ELEMENT_IDS);
if (value instanceof List<?>) {
    List<?> values = (List<?>) value;
    List<Object> objects = values.stream()
        .filter(String.class::isInstance)
        .map(String.class::cast)
        .map(id -> this.objectService.getObject(editingContext, id))
        .collect(Collectors.toList());
}
```